integration_test/README.md

```diff
@@ -0,0 +1,79 @@
+
+# Integration Testing
+Allows testing lando with the exception of Openstack.
+- runs rabbitmq inside docker container
+- clones [bespin-api](https://github.com/Duke-GCB/bespin-api)
+- requires DukeDS credentials and example project
+
+## Initial Setup
+
+### Create project to hold an input file and results
+Create a project on DukeDS and upload a DNA sequence file. https://dataservice.duke.edu
+Record the project uuid, file uuid and agent/user keys.
+In a terminal in this directory(integration_test).
+Copy example.dds.config to dds.config.
+Using data from data service fill in the following variables in dds.config: 
+```
+export DDS_PROJECT_ID='__FILL_IN__' 
+export SEQ_DDS_FILE_ID='__FILL_IN__'
+export AGENT_KEY='__FILL_IN__'
+export USER_KEY='__FILL_IN__'
+```
+If running on OSX update `tmpdir-prefix` and `tmp-outdir-prefix` to cwl_base_command in `worker_config.yml`:
+```
+cwl_base_command:
+  - "cwl-runner"
+  - "--debug"
+  - "--tmpdir-prefix=/Users/<username>/tmp"
+  - "--tmp-outdir-prefix=/Users/<username>/tmp"
+```
+Otherwise you can remove the `--tmpdir-prefix` and `--tmp-outdir-prefix` tags.
+
+### Checkout bespin-api 
+In a terminal in this directory(integration_test).
+Clone bespin-api and setup config files:
+```
+./setup.sh
+```
+This should create the bespin-api directory containing a database with one job.
+ 
+
+### Start bespin-api and rabbitmq:
+In a terminal in this directory(integration_test).
+Run this:
+```
+./start.sh
+```
+This will hang until you press CONTROL-C.
+At this point you can navigate to http://127.0.0.1:8000/api/jobs/1/ and see the job.
+You will need username:lando password:secret.
+
+### Run lando
+In another terminal in the root directory of this repo:
+```
+virtualenv env
+source env/bin/activate
+pip install git+git://github.com/Duke-GCB/lando-messaging.git 
+python setup.py install 
```

Comment:
It seems to be spotty. I've created virtualenvs in the past few weeks (since I installed anaconda), but it doesn't fail on everything.